includes/binary_object_upload.form.inc

```diff
@@ -99,7 +99,7 @@ function islandora_binary_object_upload_form_submit(array $form, array &$form_st
   else {
     $datastream = $object['OBJ'];
   }
-  $file = file_load($form_state['values']['file']);
+  $file = file_load(isset($form_state['values']['file']) ? $form_state['values']['file'] : NULL);
```

Comment:
Not... sure that this will do quite what you want... possibly throwing warnings?... I think instead, we'd want to do something like:

```php
if (isset($form_state['values']['file'])) {
  $file = file_load($form_state['values']['file']);
  [ ... could possibly move lines 96 to 101 in, or drop 'em and replace, with something like: ]
  $datastream = isset($object['OBJ']) ?
    $object['OBJ'] :
    $datastream->constructDatastream('OBJ', 'M');
  [... the other stuff dealing with setting up the OBJ datastream ...]
}
```